app/src/main/java/com/portfolio/tasky/globals/DataUtils.kt

```diff
@@ -0,0 +1,30 @@
+package com.portfolio.tasky.globals
+
+import android.util.Patterns
+
+class DataUtils {
+
+    companion object{
+        object Validators{
+            var isEmailValid : Boolean = false
+            var isNameValid : Boolean = false
+            var isPasswordValid : Boolean = false
+
+            fun clearValidatorParams(){
+                isEmailValid = false
+                isNameValid = false
+                isPasswordValid = false
+            }
+
+            fun isEmailValid(emailID : CharSequence?) : Boolean{
+                return Patterns.EMAIL_ADDRESS.matcher(emailID as CharSequence).matches()
+            }
+            fun isPasswordValid(password : CharSequence) : Boolean{
+                return Regex("^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z]).{9,}\$").matches(password)
+            }
+            fun isNameValid(name : CharSequence) : Boolean{
+                return Regex("^(?=.*[a-z]).{3,}\$").matches(name)
```

Comment:
These can all be usecases
Will make more sense, since each would be in their own class (Check the video about SOLID in the video portal of the mentorship, the S specifically)

Also, using Patterns.EMAIL_ADDRESS will make it so the method isn't Unit Testable since it's an Android-specific dependency. You can check the import in like 3 "import android.util.Patterns"
I recommend:

1. You create an interface
2. You write an implementation of that, which uses Patterns.EMAIL_ADDRESS
3. You pass the abstraction to the constructor of your class needing it(The ValidateEmailUseCase), or if you want to keep this class, then you can pass it on the constructor or the parameter

An example:

```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

```
class EmailPatternValidatorImpl: EmailPatternValidator {
    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Reply:
Thank you for explaining in this detail. Your explanation helped me implement these use-cases.